src/controllers/krate/search.rs

```diff
@@ -338,28 +338,16 @@ pub fn search(req: &mut dyn RequestExt) -> EndpointResult {
                 )
             },
         )
-        .collect();
-
-    #[derive(Serialize)]
-    struct R {
-        crates: Vec<EncodableCrate>,
-        meta: Meta,
-    }
-    #[derive(Serialize)]
-    struct Meta {
-        total: Option<i64>,
-        next_page: Option<String>,
-        prev_page: Option<String>,
-    }
+        .collect::<Vec<_>>();
 
-    Ok(req.json(&R {
-        crates,
-        meta: Meta {
-            total: Some(total),
-            next_page,
-            prev_page,
+    Ok(req.json(&json!({
+        "crates": crates,
+        "meta": {
+            "total": Some(total),
```

Comment:
@Turbo87 So, do we need to simplify here? It also looks like it will always be Some().

Reply:
good catch! :)

we don't necessarily *need* to simplify it, but I guess we should. I've rebased and fixed it in the process.